Dockerfile-xcompiler

```diff
@@ -55,4 +51,3 @@ ADD cross-build/ /home/xcompiler/webthree-umbrella/cross-build/
 
 # Build the cross-compiler
 WORKDIR /home/xcompiler/webthree-umbrella/cross-build/ct-ng
-RUN ./ct-ng.sh ~/ct-ng $TARGET_ARCHITECTURE "1.20.0"
```

Comment:
newer versions of docker seem to have an [ARG](http://docs.docker.com/engine/reference/builder/#arg) command. unless we update to it, we won't be able to parameterize the docker file it seems, so I went with [this suggestion](https://github.com/docker/docker/issues/14634#issuecomment-136793152) instead, it's pretty clean IMO